platform/lib/pipeline/componentReferenceImporter.js

```diff
@@ -72,25 +74,42 @@ class ComponentReferenceImporter {
     // Keep track of all promises to complete function
     const savedDocuments = [];
 
+    const versionMapping = {};
     for (const extension of extensions) {
       const documents = await this._findExtensionDocs(extension);
       const versions = [...new Set(documents.map((doc) => doc.version).sort().reverse())];
 
+
       if (!documents.length) {
         log.warn(`No matching document for component: ${extension.name}`);
       } else {
+        const supportedVersions = {};
+        versionMapping[extension.name] = supportedVersions;
         documents.forEach((doc) => {
           // IMPORTANT: first rewrite URLs
           this._rewriteRelativePaths(extension.path, doc.document);
           // then set versions to avoid re-writing the relative version toggle links
           this._setMetadata(
               doc.tagName || extension.name, doc.document, doc.version, versions);
+          if (doc.document.isCurrent) {
+            supportedVersions.current = doc.version;
+          }
+          doc.document.stripInlineTitle();
           savedDocuments.push(
               this._saveDocument(doc.tagName || extension.name, doc.document, doc.version));
+          doc.document.formats.forEach((format) => {
+            let versionsPerFormat = supportedVersions[format];
+            if (!versionsPerFormat) {
+              versionsPerFormat = [];
+              supportedVersions[format] = versionsPerFormat;
+            }
+            versionsPerFormat.push(doc.version);
+          });
```

Comment:
I was a little confused by this block (line 100-107), I didn't test this but you can probably turn it into a one liner like so (Sets are nifty because their values are automatically unique, and the add method returns the set itself):

```
supportedVersions[format] = (supportedVersions[format] || new Set()).add(doc.version);
```

Reply:
Unfortunately, doesn't work as JSON serialization requires an array.

Reply:
`Array.from(supportedVersions[format])` :) no big deal though.